### ISSUE TITLE
Initial delay when getting S3 object using AWS Java SDK v2 async methods

### ISSUE THREAD

**Issue by Halmaethor** [localstack/localstack#3096]
# Type of request: This is a ...
[x] bug report
[ ] feature request

# Detailed description
Using the AWS Java SDK v2,  when getting a S3 object after an SQS operation (eg: purgeQueue), the s3 operation take at least 30 seconds when using the async method, but it takes less than a second when using the sync method.

If the SQS operation is removed, both s3 operation take approximately the same time.

We also used Wireshark to inspect the HTTP requests and we saw that the main difference between the async and sync requests is "Connection: Keep-Alive" header parameter exists for the sync method, but not for the async method.

## Expected behavior
We expect that both methods, async and sync, should take approximately the same time.

## Actual behavior
The sync method is really fast while the async method take at least 30 seconds.

# Steps to reproduce
Assuming that:

* a SQS queue with name "queueName" exists
* a S3 bucket with name "bucketName" exists
* a non-empty S3 object with name "filePath" exists

Execute the provided snippet on a Kotlin IDE and validate the different run times between the async and sync methods.

## Command used to start LocalStack
Via docker container:

```
localstack:
    image: localstack/localstack:0.11.6
    container_name: localstack
    ports:
      - 4566:4566
      - 8080:8080
    environment:
      - SERVICES=s3,sns,sqs
      - DEFAULT_REGION=eu-west-1
      - DATA_DIR=/tmp/localstack/data
      - DEBUG=1
    volumes:
      - '~/.localstack:/tmp/localstack'
      - '/var/run/docker.sock:/var/run/docker.sock'
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
```kotlin
fun test() = runBlocking {
        val sqsClient = SqsAsyncClient.builder()
            .endpointOverride(URI("http://localhost:4566"))
            .build()

        val sqsResponse = sqsClient.purgeQueue(
            PurgeQueueRequest.builder().queueUrl("http://localhost:4566/000000000000/queueName").build()
        ).await()

        downloadSync()
        downloadAsync()
}

private suspend fun downloadAsync() {
        val s3Client = S3AsyncClient.builder()
            .endpointOverride(URI("http://localhost:4566"))
            .build()
        var startTime = Instant.now()
        val fileContent = s3Client.getObject(
            GetObjectRequest.builder().bucket("bucketName").key("filePath").build(),
            AsyncResponseTransformer.toBytes()
        ).await().asString(Charset.defaultCharset())
        val endTime = Instant.now()
        println("Async download took: ${Duration.between(startTime, endTime)}")
}

private fun downloadSync() {
        val s3Client = S3Client.builder()
            .endpointOverride(URI("http://localhost:4566"))
            .build()
        var startTime = Instant.now()
        val fileContent = s3Client.getObject(
            GetObjectRequest.builder().bucket("bucketName").key("filePath").build()
        ).readAllBytes()!!.contentToString()
        val endTime = Instant.now()
        println("Sync download took: ${Duration.between(startTime, endTime)}")
}
```

┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-287) by [Unito](https://www.unito.io/learn-more)

**Comment by HubertLegec** [comment]
I'm facing similar issue with localstack 0.11.6. Spring Boot tests which use AWS Java SDK v2 and async clients (for S3, SNS and SQS) are significantly slower comparing to localstack 0.11.2.

**Comment by workmanw** [comment]
Ditto. We also ran into this issue and can't make heads or tails of it.

**Comment by whummer** [comment]
Thanks for reporting @Halmaethor @HubertLegec @workmanw . I'd assume that the performance issue you were experiencing was potentially fixed in #3091 . A new version `0.12.0` has been tagged in #3118 - can you please give it another try with the latest version and let us know if the problem persist? Thanks

**Comment by workmanw** [comment]
@whummer I'm sorry for never getting back on this. Yes, our perf issues were solved. I didn't originally report it, but this issue matched mine and is fixed for me.
